### DISCUSSION TITLE
Estimating commit worth

### DISCUSSION THREAD

**Discussion by abitrolly** [mindsdb/mindsdb#2303] [General]
I wonder if MindsDB will be able to predict commit effort given a PR?

The particular problem that bugs me is that commit content is a big unstructured data, that doesn't fit nicely into SQL tables.

**Comment by paxcema** [comment]
Hi @abitrolly,

This is an interesting problem. Because MindsDB supports only supervised learning (for now, at least), it would need access to some sort of metric or label that describes the "effort" behind a PR. "Effort" as a concept in this context would also have to be defined, in the first place. In my opinion, this is the biggest challenge.

After that, access to the repository and subsequent diffs that each PR introduce would have to be passed to the predictor when making a query. I am also certain that this task would require a custom mixer, because every query would be referencing and comparing against the latest "stable" version of the repo. As for how the actual SQL table would look, maybe it's enough to store commit hashes and the origin repo, so that the model/mixer internally fetches the actual code.

**Reply by abitrolly** [reply]
People will always try to outplay the game, so the "effort" needs to be explanatory. It will be changeable from issue to issue. MindsDB could help to estimate the effort, leaving to decide on final weights to people.

Here is a discussion that hopefully makes the goal more clear https://gitlab.com/gitlab-org/gitlab/-/issues/327138
